### ISSUE TITLE
Interpolation for components in selectors 

### ISSUE THREAD

**Issue by peduarte** [modulz/stitches#69]
The ability to do something like this:

```jsx
const Icon = styled("span", {
  color: "blue"
});

const ButtonWithIcon = styled('button', {
  ":hover span": {
    color: "red"
  },

  [`:hover ${Icon}`]: {
    color: "red"
  }
});
```

**Comment by christianalfoni** [comment]
This should be pretty straight forward.

In `styled` create a sequence that produces unique component classnames for each type of instance.

Then add a `toString` to each instance which returns that name.

When the component renders its classname, include this unique component classname as well.

```ts
className: css(componentClassName, ...compositions, props.className)
```

**Comment by hadihallak** [comment]
yeah, I think that should do but instead of using a sequence, I think it would be safer to create a hash based on the styles, variants & tag name of a component. this way, even if components got instantiated in a different order on the server and the client, it would still have the same hashed class
